### PR TITLE
Add Load More for Plugin Directory

### PR DESCRIPTION
**Pull request by lachlanhardy**
**This PR is a Work-In-Progress. DO NOT MERGE.**

This PR adds the Load More functionality for the Plugins directory.

The work we know about:
- [x] Add Load More link
- [x] Write some JS to usefully load the rest of the plugins
- [x] Style Load more
- [x] refactor JS so it's readable/efficient etc
- [x] Make the page use the NextJS Link component
- [x] Load all plugins and hide via CSS
- [x] replace URL call with straight JS
- [x] Add a nice transition for the load
- [x] Check responsive layouts and styles in different browsers

### CODE REVIEW

**Comment by toolmantim** [comment]
Oh interesting!! Awesome for putting this up as a WIP, as it makes me wonder of different ways to do it.

When I was designing it, I was thinking it was just hiding it using CSS. I wonder if that's a bad idea? 🤔

**Comment by lachlanhardy** [comment]
Not a bad idea at all! I think I was in JS component mode so I looked to solve it with JS. I'll take another look today.

File: pages/plugins/index.js [hunk 1]
```diff
@@ -14,7 +15,7 @@ import image from '../../assets/images/plugins/plugin.png'
 
 import { all as allPlugins, mostStarred, newlyAdded, recentlyUpdated } from '../../lib/data/plugins'
 
-const Link = styled.a`
+const PluginLink = styled(Link)`
```
Comment (toolmantim) [inline pages/plugins/index.js]: I think the `<Link>` component (both ours, and Next's) is a HOC designed to wrap around an actual `<a>` tag, and it passes down the `href` attribute to it. I made it all super confusing when I called the anchor StyledComponent on this page `Link` too. Sorry!!

So perhaps in our case, we should rename `Link` to `TextLink`:

```suggestion
const TextLink = styled.a`
```

You'd then use it with `Link` like so:

```jsx
<!-- Link to external site -->
<Link href="https://somesite.com/" external>
  <TextLink>Some site</TextLink>
</Link>

<!-- Internal link -->
<Link href="/some-page">
  <TextLink>Some page</TextLink>
</Link>
```

And there's no need to use `<Link>` at all for external links on this page… we could get away with just using it for the load more button? And leave the rest as `TextLink`s?

Comment (lachlanhardy) [reply pages/plugins/index.js]: Good call. Doneski.

File: pages/plugins/index.js [hunk 2]
```diff
@@ -176,79 +195,99 @@ export default page((props) => (
...
+            </TextLink>
           </Title>
           <Description>{plugin.description}</Description>
           <Ellipsis>
             <Repo>
-              <Link href={pluginUrl({ plugin })}>
-                {`github.com/${plugin.owner.login}/${plugin.repo}`}
-              </Link>
+              <TextLink href={pluginUrl({ plugin })}>
+                <>
```
Comment (toolmantim) [inline pages/plugins/index.js]: Don't need these `<>` anymore either!

Comment (lachlanhardy) [reply pages/plugins/index.js]: I feel like I'm saying "good catch" too often. I'll work on that.

I guess this is why reviews are good!

Comment (toolmantim) [reply pages/plugins/index.js]: All good! Happy to collab on stuff :)

**Pull request merged**